core/domain/html_validation_service_test.py

```diff
@@ -1512,7 +1514,118 @@ def test_svg_string_validation(self):
             ' -595 -23 -112 -14 -299 21 -402 70 -206 212 -368 402 -460 118 -56'
             ' 181 -69 339 -69 104 0 156 4 204 18 255 71 451 258 537 511 35 103'
             ' 44 290 21 402 -60 278 -258 497 -532 586 -88 28 -324 33 -420 9z"/'
+            '> </g> </svg>')
+        self.assertEqual(
+            html_validation_service.get_invalid_svg_tags_and_attrs(
+                invalid_svg_string), (['testtag'], []))
+        # SVG containing an invalid attribute for a valid tag.
+        invalid_svg_string = (
```

Comment:
Perhaps simplify these tests a bit -- you can drop some of the long tags that are less relevant rather than bury things in large strings. Otherwise, hard for a human to easily grok exactly what's wrong with the string.